pyproject.toml

```diff
@@ -1,261 +1,266 @@
-    "astropy", # Used for accounting leap seconds when converting epoch timestamps
-    "pyPublicDecompWt"  # Required to read SEVIRI data
-]
+[build-system]
+requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning>=1.0.0,<2.0.0"]
+build-backend = "poetry_dynamic_versioning.backend"     # object used to perform the build process
 
-[project.optional-dependencies]
-test = [
-    "pytest",  # Required for unit tests
-    "pytest-cov",  # Reports on test coverage
-    "pixelmatch"  # Reports on test coverage
-]
+[tool.poetry.dependencies]     # must download to run
+python = ">=3.9"    # mandatory to declare the required python version
+matplotlib = ">=3.7.0"     # Base requirement works, version specific to test outputs
+netcdf4 = "*"              # Base requirement
+numpy = "*"                # Base requirement
+pyresample = "*"           # Base requirement Geospatial image resampling
+			     	       # efficiency improvements >= 1.22.3
+pyyaml = "*"               # Base requirement
+scipy = "*"  	           # Base requirement, requires openblas
+xarray = "*"               # Base requirement
+astropy = "*"              # Used for accounting leap seconds when converting epoch timestamps
+cartopy = ">=0.22.0"         # No longer requires libgeos. 0.21 incompatible with geos 3.12
+ephem = "*"                # Required for overpass predictor
+h5py = "*"                 # hdf5 readers (GMI)
+isodate = "*"              # Required for overpass predictor
+jsonschema = ">4.18.0"   # Provides validation for YAML plugins
+numexpr = "<2.8.5"         # for efficiency improvements
+psutil = "*"               # Required for memory checks
+pyhdf = "*"                # hdf4 readers (MODIS)
+pyaml-env = "*"            # Reading YAML output config files, with paths
+pyorbital = "*"            # required by satpy
+pypublicdecompwt = "*"     # Required to read SEVIRI data
+rasterio = "*"             # GEOTIFF output
+referencing = "*"
+satpy = "*"               # efficiency improvements >= 0.33.1
+scikit-image = "*"        # Radius based center coverage checks
+
+# Optional dependencies
+# To add a new optional dependency it must be added both here and in one of the "extra"
+# groups below.
+#
+# Doc group
+pydata-sphinx-theme = { version = "*", optional = true }
+sphinx = { version = "<7.2", optional = true }
+sphinx-design = { version = "*", optional = true }
+# Lint group
+bandit = { version = "*", optional = true }
+black = { version = "*", optional = true }
+pylint = { version = "*", optional = true }
+flake8 = { version = "*", optional = true }
+flake8-rst-docstrings = { version = "*", optional = true }
+flake8-docstrings = { version = "*", optional = true }
+flake8-rst = { version = "*", optional = true }
+prettier = { version = "*", optional = true }
+doc8 = { version = "*", optional = true }
+# Test group
+pytest = { version = "*", optional = true }
+pytest-cov = { version = "*", optional = true }
+pixelmatch = { version = "*", optional = true }
```

Comment:
Did you catch all the updated versions / packages since this was branched? @jsolbrig I think it was just jsonschema, numexpr, and pixelmatch?

Reply:
Yes, I was careful to go through and capture everything. It might be good for you to double-check but I'm 99% sure I got everything.

Reply:
adding from what I did from my end :  numexpr was recently updated (this week) with a constraint <2.8.5 when my build failed. Jeremy asked for this change.